datafusion/core/src/datasource/file_format/parquet.rs

```diff
@@ -151,12 +152,12 @@ impl FileFormat for ParquetFormat {
         store: &Arc<dyn ObjectStore>,
         objects: &[ObjectMeta],
     ) -> Result<SchemaRef> {
-        let mut schemas = Vec::with_capacity(objects.len());
-        for object in objects {
-            let schema =
-                fetch_schema(store.as_ref(), object, self.metadata_size_hint).await?;
-            schemas.push(schema)
-        }
+        let schemas: Vec<_> = futures::stream::iter(objects)
+            .map(|object| fetch_schema(store.as_ref(), object, self.metadata_size_hint))
+            .boxed() // Workaround https://github.com/rust-lang/rust/issues/64552
+            .buffered(32)
```

Comment:
I think the `32` should at least be a named constant so it is more discoverable

What do you think about potentially using the same value as in https://github.com/apache/arrow-datafusion/pull/6183:

```
const CONCURRENCY_LIMIT: usize = 100;
```

(I sort of imagine some day someone will want to make that a configuration knob rather than a constant so using the same constant in the code will make it easier to find where they are used)

Reply:
I'll make this change as a follow up

Reply:
https://github.com/apache/arrow-datafusion/pull/6389